docs(autofix): clarify when Autofix runs automatically (#17649)
getsentry/sentry-docs: docs/product/ai-in-sentry/seer/autofix/index.mdx

2. [Solution Identification](#solution-identification): Identify a solution based on the analysis
3. [Code Generation](#code-generation): Generate a code fix and (optionally) open a PR, or hand off to a coding agent

You can always manually trigger the Autofix flow from the Issue Details page.

### When Autofix Runs Automatically

When automation is enabled in your project's Seer settings, Sentry can also trigger Autofix automatically on issues that meet the following criteria:

1. **The issue has 10 or more events** — Sentry waits until the issue has captured at least 10 events to ensure it's worth investigating.
2. **The issue occurred within the last 14 days** — Only recent issues are eligible for automatic runs.
3. **The issue has a sufficient fixability score** — Sentry's ML model must estimate the issue is reasonably fixable (see below).

#### How Far Autofix Goes

> How far Autofix proceeds automatically depends on two factors:

Reviewer: ```suggestion
How far Autofix automatically goes depends on two factors:
```

1. **Fixability confidence** — Sentry's ML model evaluates each issue and estimates how straightforward it is to resolve. Issues with higher confidence may proceed through more automation steps.

> 2. **Your automation preferences** — In your project's Seer settings, you can cap how far automation goes, regardless of the fixability confidence.

Reviewer: ```suggestion
2. **Your automation preferences** — In your project's [Seer settings](https://sentry.io/settings/seer/projects), you can cap how far automation goes, regardless of the fixability confidence.
```

> 2. **Your automation preferences** — In your project's [Seer settings](https://sentry.io/settings/seer/projects), you can cap how far automation goes, regardless of the fixability confidence.

Reviewer: ```suggestion
2. **Your automation preferences** — In your project's [Seer settings](https://sentry.io/orgredirect/organizations/:orgslug/settings/seer/projects), you can cap how far automation goes, regardless of the fixability confidence.
```

> The automation preference acts as a maximum stopping point. Even if an issue has high fixability confidence, Autofix will respect your configured limit. The available stopping points are:

Reviewer: ```suggestion
Your automation settings decide the stopping point. Even if an issue has high fixability confidence, Autofix will only automate as far as you've set it. The available stopping points are:
```

- **Stop after Root Cause** — Autofix identifies the cause but doesn't suggest code changes.
- **Stop after Plan** — Autofix suggests a fix but doesn't open a PR.
- **Stop after PR Drafted** — Autofix can open a PR to apply the suggested fix.

> For example, if you've set **Stop after Plan**, Autofix will perform root cause analysis and solution planning, but won't create a PR—even for issues with very high fixability confidence. Conversely, if fixability confidence is low, Autofix may stop at root cause analysis even if your settings allow it to proceed further. 

Reviewer: ```suggestion
For example, if you've set **Stop after Plan**, Autofix will perform root cause analysis and solution planning, but won't create a PR, even for issues with very high fixability confidence. 
```

2. **The issue occurred within the last 14 days** — Only recent issues are eligible for automatic runs.
3. **The issue has a sufficient fixability score** — Sentry's ML model must estimate the issue is reasonably fixable (see below).

#### Fixability Score

When Seer evaluates an issue, it assigns a **fixability score** — a value between 0 and 1 that estimates how straightforward the issue is to resolve. This score is generated by a machine-learning model that analyzes the issue summary and classifies it on a spectrum from "complex and very difficult to resolve" to "simple and easily resolved."

The fixability score determines **whether** Autofix runs automatically, and **how far** it goes:

- **Below 0.40 (Low)** — The issue is not considered fixable enough for automation. Autofix will not run automatically, but you can still trigger it manually.
- **0.40 – 0.66 (Medium)** — Autofix runs automatically but stops at **root cause analysis**. Sentry identifies the likely cause and presents it for your review.
- **0.66 – 0.78 (High)** — Autofix runs automatically through **code changes**. Sentry identifies the root cause and suggests a fix.

> - **0.78 and above (Very High)** — Autofix runs automatically and can **open a pull request** with the suggested fix.

Reviewer 2: I think we will proceed _up to_ these stopping points based on the score, but if the users preference is for less then we'll stop earlier. 

Example: say the user has set "Automation Steps: Stop after Plan"
- if the score is 0.99 then we'll do Root Cause and Planning, but not PR create because the user limits it.
- if however the score was 0.50 then we'd do the Root Cause Step, but not go on to do a Plan, because the score is too low.

That's my understanding anyway. 

---

Also, the labels in here don't match with the product. In product we've got: 
- Stop after Root Cause
- Stop after Plan
- Stop after PR Drafted

Reviewer: I'm not sure we need to provide this level of detail unless we expose the score. @ryan953 - do we plan to share any further detail about a fixability score or the fixability confidence of an issue in the autofix UX?

Reviewer 2: i think also fixability score is kind of being replaced by the night-shift thing? idk. we probably shouldn't be so specific because its outside the control of users, and will likely change. 

for example, i would _love_ for the 10 event minimum to change so i can get automatic fixes for flakey tests sooner

Reviewer 2: @cursor fixup this section based on the thread here

#### Capping Automation Depth

Even when the fixability score suggests a deeper level of automation, you can set a maximum stopping point in your project's Seer settings. The available options are:

- **Root cause analysis only** — Autofix identifies the cause but doesn't suggest code changes.
- **Code changes** — Autofix suggests a fix but doesn't open a PR.
- **Open a pull request** — Autofix can go all the way to opening a PR (default).

> Your chosen stopping point acts as an upper bound — Autofix will never go further than what you allow, regardless of the fixability score.

Reviewer: ```suggestion
Even when the fixability score suggests next steps in automation, you can set an earlier stopping point in your project's Seer settings. The available options are:

- **Root cause analysis only** — Autofix identifies the cause but doesn't suggest code changes.
- **Code changes** — Autofix suggests a fix but doesn't open a PR.
- **Open a pull request** — Autofix can open a PR to apply the suggested fix.

Selecting to stop any any of these points will prevent Seer from automatically proceeding. 
```
